welly/curve.py

```diff
@@ -1,273 +1,150 @@
+                 data,
+                 index=None,
+                 mnemonic=None,
+                 dtype=None,
+                 api=None,
+                 basis_units=None,
+                 code=None,
+                 description=None,
+                 date=None,
+                 index_name=None,
+                 null=None,
+                 run=None,
+                 service_company=None,
+                 units=None,
+                 start=None,
+                 stop=None,
+                 step=None):
         """
-        obj = np.asarray(data).view(cls).copy()
-
-        params = params or {}
-
-        for k, v in params.items():
-            setattr(obj, k, v)
-
-        if basis is not None:
-            if basis[0] > basis[1]:
-                basis = np.flipud(basis)
-            setattr(obj, 'start', basis[0])
-            setattr(obj, 'step', basis[1]-basis[0])
-
-        return obj
+        Args:
+            data (ndarray (structured or homogeneous), Iterable, dict, or DataFrame):
+                1D/2D curve numerical or categorical data. Dict can contain Series, arrays, constants, dataclass or
+                list-like objects. If data is a dict, column order follows insertion-order. Input is passed to 'data'
+                parameter of pd.DataFrame constructor.
+            index (Index or array-like):
+                index to use for resulting pd.DataFrame. Will default to RangeIndex if no indexing information part of
+                input data and no index provided. Input is passed to 'index' parameter of the pd.DataFrame constructor.
+            mnemonic (str):
+                the mnemonic of the curve if the data does not have them. It is passed as the 'columns' param of
+                pd.DataFrame constructor.
+            dtype (str):
+                data type to force. Only a single dtype is allowed. If None, infer. Passed to pd.DataFrame constructor.
+            api (str):
+                application program interface number.
+            basis_units (str):
+                unit of the index (e.g. 'ft', 'm', 'ms').
+            code (int):
+                log code
+            date (str):
+                date of when the curve was recorded.
+            description (str):
+                description of the curve.
+            index_name (str):
+                name of the index that will be assigned to pd.DataFrame.index.name (e.g. 'depth', 'time').
+            null (float):
+                numeric null value representation (e.g. -9999).
+            run (int):
+                the count of the run of the same measurement through the same well.
+            service_company (str):
+                company that executed logging operations.
+            start (float):
+                index value where index starts
+            step (float):
+                index value increment
+            stop (float):
+                index value where index stops
+            units (str):
```

Comment:
I thought the same but welly currently has it defined as 'units'. The reason I defined 'units' instead of 'unit' is because we would like to keep the interface the same as much as possible.